### PR TITLE
Minor issues to clear the ground for dual fields and reaction force calculation for static linear elasticity

### PR DESCRIPTION
**Pull request by kmokstad**
OPM/IFEM-Elasticity#39 depends on this one. Apart from that, nothing here should be controversial.

### CODE REVIEW

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=39 please

File: src/SIM/SIMbase.C [hunk 1]
```diff
@@ -1028,11 +1034,11 @@ bool SIMbase::solveSystem (Vector& solution, int printSol, double* rCond,
...
                                  const char* compName)
 {
-  solution.resize(myEqSys->getNoRHS());
+  solution.resize(this->getNoRHS());
   for (size_t i = 0; i < solution.size(); i++)
     if (!this->solveSystem(solution[i],printSol,nullptr,compName,i==0,i))
       return false;
-    else
+    else if (solution.size() > 2)
       printSol = 0; // Print summary only for the first solution
```
Comment (akva2) [inline src/SIM/SIMbase.C]: comment seems wrong.

**Comment by akva2** [comment]
and i don't really understand the first fix, but i trust it's doing something.

**Comment by akva2** [comment]
you will get a failure in elasticity after the LR merge. those pr's updating the regtests was stale it seems (same thing happens in poisson..)

**Comment by kmokstad** [comment]
I know, probably I screwed up something when rebasing Kjetil's PRon the LR things. Thought it should be equivalent code, but apparently not :(

Regarding the first commit, it is only (afaik) affects the console printout for problems with multiple material regions. It then printed ca code value for the material that did not exist (typically 10000), and it caused be some confusion and lost time debugging ghosts. This fixes such that the value printed is the value stored in the Property container.

**Comment by akva2** [comment]
personally i think it's just as likely the PR's were stale. i will check.

**Comment by akva2** [comment]
yup. stale pr's, you didn't screw up anything.

**Comment by akva2** [comment]
since you had some other things in the queue that affects those tests, do the update in elasticity with that or the downstream of this ?

**Comment by kmokstad** [comment]
I know that the exact_p[12] elasticity fails, due to the inclusion of the external energy from the reaction forces (+ the adaptive failures, which then is not due to me (phew)). But I just wanted to see if there were other downstream issues, therefore this Jenkins job. I will fix the exact reg-test when I'm sure the new values are the correct ones (not on the top of priority just now though).

**Comment by akva2** [comment]
then i will fix the disc-stair one - i need trustworthy CI.

File: src/SIM/SIMbase.C (hunk 1 above)
Comment (kmokstad) [reply src/SIM/SIMbase.C]: Indeed, now it reads "first two solutions" (the second one being the upcoming dual solution)

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-poisson#20 ifem-elasticity=41 ifem-darcy#10 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-poisson=20 ifem-elasticity=41 ifem-darcy=10 please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-poisson=20 ifem-elasticity=41 ifem-darcy=10 please

**Pull request merged**